netbox/utilities/versions.py

```diff
@@ -0,0 +1,35 @@
+import requests
+from cacheops import cached
+from django.conf import settings
+from packaging import version
+
+
+@cached(timeout=settings.GITHUB_VERSION_TIMEOUT if settings.GITHUB_VERSION_TIMEOUT > 0 else 1)
+def get_releases(pre_releases=False):
+    url = 'https://api.github.com/repos/{}/releases'.format(settings.GITHUB_REPOSITORY)
+    headers = {
+        'Accept': 'application/vnd.github.v3+json',
+    }
+    try:
+        response = requests.get(url, headers=headers)
+        releases = [(version.parse(release['tag_name']), release.get('html_url'))
+                    for release in response.json()
+                    if 'tag_name' in release]
+    except Exception:
+        releases = []
+
+    if not pre_releases:
+        releases = [(release, url)
+                    for release, url in releases
+                    if not release.is_devrelease and not release.is_prerelease]
```

Comment:
This conditional logic can be moved up to avoid looping through the list of releases twice:

```python
if 'tag_name' in release and (not release.is_devrelease and not release.is_prerelease) or pre_releases
```

However, it's probably worth ditching list comprehension in favor of a regular `for ... in` stanza for the sake of clarity.

Reply:
The result is cached for hours, so iterating over the list twice is negligible. I personally feel that list comprehension is clearer than a loop, but that's personal.